### PR TITLE
Fix: refactorization of "getListForItem" function

### PR DESCRIPTION
**Pull request by hregis**
Fix https://github.com/Dolibarr/dolibarr/issues/7549

### CODE REVIEW

File: htdocs/product/class/api_products.class.php [hunk 1]
```diff
@@ -263,9 +263,13 @@ function delete($id)
      *
      * @url GET {id}/categories
      */
-    function getCategories($id, $sortfield = "s.rowid", $sortorder = 'ASC', $limit = 0, $page = 0) {
-        $categories = new Categories();
-        return $categories->getListForItem($sortfield, $sortorder, $limit, $page, 'product', $id);
+	function getCategories($id, $sortfield = "s.rowid", $sortorder = 'ASC', $limit = 0, $page = 0)
+	{
+		require_once DOL_DOCUMENT_ROOT.'/categories/class/api_categories.class.php';
```
Comment (eldy) [inline htdocs/product/class/api_products.class.php]: Why this require.
Do you mean  require_once DOL_DOCUMENT_ROOT.'/categories/class/categories.class.php';   instead ?

Comment (hregis) [reply htdocs/product/class/api_products.class.php]: @eldy 
the function "getListForItem" is in "api_categories.class.php"... i move this function in categorie.class.php ?

Comment (hregis) [reply htdocs/product/class/api_products.class.php]: @eldy 
if i move the function "getListForItem" in "categorie.class.php" i can't clean the data with "_cleanObjectDatas" !

Comment (hregis) [reply htdocs/product/class/api_products.class.php]: @eldy 
ok it's fixed with new "getListForItem" function in categorie class

**Pull request merged**